model.go

```diff
@@ -377,3 +377,87 @@ func MapSetFields(originModel Model, destinationModel Model, mappings map[field.
 		}
 	}
 }
+
+func getAllFields(model interface{}) map[field.Name]reflect.Value {
+	fields := map[field.Name]reflect.Value{}
+
+	// Value of model
+	modelValue := reflect.ValueOf(model)
+	if modelValue.Kind() == reflect.Ptr {
+		modelValue = modelValue.Elem()
+	}
+
+	// Type of Model
+	modelType := reflect.TypeOf(model)
+	if modelType.Kind() == reflect.Ptr {
+		modelType = modelType.Elem()
+	}
+
+	for i := 0; i < modelType.NumField(); i++ {
+		v := modelValue.Field(i)
```

Comment:
@cleitonmarx  Not sure if this will have an impact, but I was thinking about allocating the Map.
```go
numField := modelType.NumField()
fields := make(map[field.Name]reflect.Value, numField)

for i := 0; i < numField; i++ {
		v := modelValue.Field(i)
```